src/frontend/src/optimizer/rule/pull_up_correlated_predicate_agg_rule.rs

```diff
@@ -185,21 +178,26 @@ impl Rule for PullUpCorrelatedPredicateAggRule {
 
             // Shift the top project expressions to the right by apply_left schema len, because it is used to check null-rejected by the top filter.
             let apply_left_schema = apply_left.schema().len();
+            let mut top_proj_all_null = true;
             let mut top_proj_null_bitset =
                 FixedBitSet::with_capacity(top_project.base.schema().len() + apply_left_schema);
             for (i, expr) in top_proj_exprs.iter().enumerate() {
                 if Strong::is_null(expr, agg_null_bitset.clone()) {
                     top_proj_null_bitset.insert(i + apply_left_schema);
+                } else {
+                    top_proj_all_null = false;
```

Comment:
When `Strong::is_null` return false, it means we can't get any useful information, so we shouldn't allow apply rule based on it.